modules/rest/src/main/java/com/enonic/xp/app/contentstudio/rest/resource/content/ContentResource.java

```diff
@@ -570,7 +572,22 @@ public TaskResultJson delete( final DeleteContentJson params )
 
     @POST
     @Path("getDependencies")
-    public GetDependenciesResultJson getDependencies( final ContentIdsJson params )
+    public GetDependenciesResultJson getDependencies( final GetDependenciesJson params )
+    {
+        final boolean isMasterQuery = ContentConstants.BRANCH_MASTER.equals( params.getBranch() );
+        final Branch branch = isMasterQuery ? ContentConstants.BRANCH_MASTER : ContentConstants.BRANCH_DRAFT;
+
+        return ContextBuilder.from( ContextAccessor.current() ).branch( branch ).build().callWith( () -> {
+            final Map<String, DependenciesJson> result = isMasterQuery ? ContextBuilder.from( ContextAccessor.current() )
+                .branch( ContentConstants.BRANCH_MASTER )
+                .build()
+                .callWith( () -> doGetDependencies( params ) ) : doGetDependencies( params );
```

Comment:
"callWith( () -> doGetDependencies( params ) ) : doGetDependencies( params );"

There is should be one call of "doGetDependencies"